src/parser/shared/modules/items/GemChecker.tsx

```diff
@@ -29,216 +32,170 @@ class GemChecker extends Analyzer {
-    const hasMaxGem = this.hasMaxGemCount(item, slot);
-    let qualitativePerformance = QualitativePerformance.Fail;
-    if (hasMaxGem) {
-      qualitativePerformance = QualitativePerformance.Good;
+  boxRowPerformance(
+    item: EventItem,
+    recommendedGems: number[] | undefined,
+    slotNumber: number,
+    slotName: JSX.Element,
+  ) {
+    let equipmentPerformance = EquipmentPerformance.Potential; //Every Item starts with Potential
+    const gemRank: { gemPerformance: EquipmentPerformance; gem: EventGem }[] = [];
+    const tooltipContent: JSX.Element[] = [];
+    const gemArrayLength = item.gems?.length ?? 0;
+    let tooltip: JSX.Element;
+
+    //Something made it that shouldn't.  Throw a filter here for those without gem or gem potential
+    if (
+      (item.bonusIDs === undefined && gemArrayLength === 0) ||
+      (!eventItemHasGemSocket(item) &&
+        !GemChecker.twoAddableGemSlots.includes(slotNumber) &&
+        !GemChecker.oneAddableGemSlot.includes(slotNumber) &&
+        gemArrayLength === 0)
+    ) {
+      tooltip = <Trans id="shared.GemChecker.NotGemable">Your {slotName} cannot take a gem.</Trans>;
+      return { equipmentPerformance, gemRank, tooltip };
     }
 
-    let maxGem = true;
-    const gemRank: { qualitativePerformance: QualitativePerformance; gem: EventGem }[] = [];
+    let allRecommendedGem = true;
 
-    if ((item.gems ?? []).length > 0) {
+    if (gemArrayLength > 0) {
       item.gems!.forEach((iGem) => {
-        const lookupGem = gemById[iGem.id];
+        const lookupGem = ITEMS[iGem.id] as CraftedItem | undefined;
 
-        let tempQP = QualitativePerformance.Fail;
+        let tempQP = EquipmentPerformance.Potential;
         const gemRec = recommendedGems?.includes(iGem.id);
 
         if (gemRec) {
-          tempQP = QualitativePerformance.Perfect;
+          tempQP = EquipmentPerformance.Perfect;
         } else {
+          allRecommendedGem = false;
           switch (lookupGem?.craftQuality) {
             case 3:
-              if (recommendedGems !== undefined && !gemRec) {
-                tempQP = QualitativePerformance.Good;
-                maxGem = false;
-              } else {
-                tempQP = QualitativePerformance.Perfect;
-              }
+              tempQP = EquipmentPerformance.Good;
               break;
             case 2:
-              tempQP = QualitativePerformance.Good;
-              maxGem = false;
+              tempQP = EquipmentPerformance.Good;
               break;
             case 1:
-              tempQP = QualitativePerformance.Ok;
-              maxGem = false;
+              tempQP = EquipmentPerformance.Ok;
               break;
-            case undefined: //Consider Item Level when there isn't a lookup.
-              tempQP = QualitativePerformance.Ok;
-              maxGem = false;
+            case undefined: //In Futuer Consider Item Level when there isn't a lookup return.
+              tempQP = EquipmentPerformance.Ok;
```

Comment:
for the time being, we should mark gems without quality as Fail. Cyrce's is already handled separately, so we don't need to worry about those missing quality.

this can help catch cases where someone has accidentally bought an old expansion gem and not realized it.
```suggestion
            case undefined: 
              tempQP = EquipmentPerformance.Fail;
```